app/scripts/controllers/createSecret.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateSecretController
+ * @description
+ * # CreateSecretController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateSecretController', function ($scope, $filter, DataService, AlertMessageService) {
+
+    $scope.newSecret = {
+      type: $scope.type === "Git Secret" ? "Source Secret" : $scope.type,
+      authType: ($scope.type === "Source Secret" || $scope.type === "Git Secret") ? "Basic Authentication" : "Credentials",
+      data: {},
+      linkSecret: false,
+      serviceAccountToLink: "",
+      linkAs: {
+        secrets: $scope.type !== 'Pull Image Secret',
+        imagePullSecrets: $scope.type === 'Pull Image Secret'
+      }
+    };
+
+    $scope.secretAuthTypeMap = {
+      "Source Secret": ["Basic Authentication", "SSH Key Based Authentication"],
+      "Pull Image Secret": ["Credentials","Docker Config"],
+      "Push Image Secret": ["Credentials","Docker Config"]
+    };
+    $scope.secretTypes = _.keys($scope.secretAuthTypeMap);
+
+    $scope.addGitconfig = false;
+    $scope.linkSecret = true;
+    $scope.serviceAccountsNames = [];
+
+    DataService.list("serviceaccounts", {namespace: $scope.namespace}, function(result) {
+      $scope.serviceAccounts = result._data;
+      $scope.serviceAccountsNames = _.keys($scope.serviceAccounts);
+    },function(result) {
+      $scope.error = {
+        message: $filter('getErrorDetails')(result)
+      };
+    });
+
+    function closeModalOrRedirect() {
+      try {
+        $scope.closeAfterCreate();
+      } catch(e) {
+        console.log('Nothing to close, redirecting');
```

Comment:
Is this trying to handle the case where it's either in a modal or a separate page? I think we can find a cleaner way to do that.